### PR TITLE
adding new announcement bar

### PR DESCRIPTION
**Pull request by smitsanghvi21**
- When `show_announcement_banner`config  is added and set to true, an announcement bar will show up above the nav.
- Message can be translated as it will use a key.
- Renders as a separate component in nav.jet just to keep it clean.

### CODE REVIEW

**Comment by smitsanghvi21** [comment]
Also pushed arabic file in this PR.

File: site/en_AU.all.json [hunk 1]
```diff
@@ -56,6 +56,7 @@
   "nav_wishlist": { "other": "My List" },
   "nav_account": { "other": "My Account" },
   "nav_signout": { "other": "Sign Out" },
+  "nav_announcement_text": { "other": "ABC Cinemas – 21st  Film Festival, June 1 – 6, 2021" },
```
Comment (stajs) [inline site/en_AU.all.json:59]:
```suggestion
  "nav_banner": { "other": "ABC Cinemas – 21st  Film Festival, June 1 – 6, 2021" },
```

File: site/styles/_nav.scss [hunk 2]
```diff
@@ -847,3 +847,15 @@ s72-dropdown,
     display: none;
   }
 }
+
+.announcement-bar {
```
Comment (stajs) [inline site/styles/_nav.scss:851]:
```suggestion
.nav-banner {
```

File: site/styles/_nav.scss [hunk 3]
```diff
@@ -847,3 +847,15 @@ s72-dropdown,
...
+  text-align: center;
+  position: relative;
+  z-index: 10;
+  background-color: $primary;
+  padding: 5px;
+  .announcement-bar-message {
```
Comment (stajs) [inline site/styles/_nav.scss:857]:
```suggestion
  .nav-banner-message {
```

File: site/templates/announcement-banner.jet [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+ {{ show_announcement_banner := config("show_announcement_banner") == "true" }}
+
+ {{ if show_announcement_banner }}
+  <div class="announcement-bar">
+    <p class="announcement-bar-message">{{i18n("nav_announcement_text")}}</p>
+  </div>
+  {{ end }}
+
+{{ end }}
```
Comment (stajs) [inline site/templates/announcement-banner.jet:11]:
```suggestion
{{block navBanner()}}

 {{showBanner := config("show_nav_banner") == "true"}}

 {{if showBanner}}
  <div class="nav-banner">
    <p class="nav-banner-message">{{i18n("nav_banner")}}</p>
  </div>
  {{end}}

{{end}}
```

File: site/templates/nav.jet [hunk 5]
```diff
@@ -1,8 +1,10 @@
 {{import "./nav/user_logged_out.jet"}}
 {{import "./nav/user_logged_in.jet"}}
+{{import "./announcement-banner.jet"}}
 
 {{block nav(showSubNav=true)}}
 <header class="navigation" data-toggle="affix">
+  {{ yield announcementBanner() }}
```
Comment (stajs) [inline site/templates/nav.jet:7]:
```suggestion
  {{yield announcementBanner()}}
```

**Review by stajs — APPROVED** [review]
Looks good, just I'm not a fan of "announcement" as it seems a bit verbose so suggest if we can make it more concise.

**Comment by smitsanghvi21** [comment]
@stajs Nav banner sounds like like a dev-friendly name. But I think from a client's point of view they might not know what "nav" means?

**Comment by stajs** [comment]
I agree, but it's the same boat as others e.g. `nav_signup`

File: site/templates/nav-banner.jet [hunk 6]
```diff
@@ -0,0 +1,11 @@
+{{block navBanner()}}
+
+ {{showBanner:= config("show_nav_banner") == "true"}}
+
+ {{if showBanner}}
+  <div class="nav-banner">
```
Comment (PeculiarGoat) [inline site/templates/nav-banner.jet:6]: Can you fix the indentation please!

**Review by PeculiarGoat — CHANGES_REQUESTED** [review]
How about 'header-banner' since this banner has nothing to do with navs :P

File: site/templates/nav-banner.jet (hunk 6 above)
Comment (smitsanghvi21) [reply site/templates/nav-banner.jet:6]: Do you mean removing the spaces?

Comment (PeculiarGoat) [reply site/templates/nav-banner.jet:6]: Our indent style is 2 spaces... looks like you have 1 space except for where you have an extra space to mis-align the end of the block.

**Comment by stajs** [comment]
@PeculiarGoat I think header-banner has some merit - I dislike that we call everything there nav. Perhaps we revisit when making these public what better names should be.

File: site/styles/_nav.scss [hunk 7]
```diff
@@ -847,3 +847,15 @@ s72-dropdown,
...
 }
+
+.nav-banner {
+  text-align: center;
+  position: relative;
+  z-index: 10;
+  background-color: $primary;
+  padding: 5px;
+  .nav-banner-message {
+      display: contents;
```
Comment (stajs) [inline site/styles/_nav.scss:858]: Why do we need this? [For Beirut I got rid of the inner element and just had the text in the `div` with reduced styles](https://github.com/shift72/beirut-dc/commit/a4dfddc9f351bce063638ff04737a952c0f95065). I want to know if I'm missing something?

Comment (smitsanghvi21) [reply site/styles/_nav.scss:858]: I think had to make few tweaks like adjusting the padding, position, etc to make it fit for the new navbar. I guess Beirut is still on the old one.

File: site/templates/header-banner.jet [hunk 8]
```diff
@@ -0,0 +1,8 @@
+{{block headerBanner()}}
+  {{showBanner:=config("show_header_banner")=="true"}}
+    {{if showBanner}}
```
Comment (PeculiarGoat) [inline site/templates/header-banner.jet:3]: This block doesn't need to be indented so far lol

File: site/styles/_nav.scss [hunk 9]
```diff
@@ -847,3 +847,15 @@ s72-dropdown,
...
+
+.header-banner {
+  text-align: center;
+  position: relative;
+  z-index: 10;
+  background-color: $primary;
+  padding: 5px;
+  .header-banner-message {
+      display: contents;
+      font-size: 13px;
```
Comment (PeculiarGoat) [inline site/styles/_nav.scss:859]: These properties are indented too far!

**Review by PeculiarGoat — CHANGES_REQUESTED** [review]

File: site/templates/header-banner.jet (hunk 8 above)
Comment (smitsanghvi21) [reply site/templates/header-banner.jet:3]: Seems to be similar with the [footer nav](https://github.com/shift72/core-template/blob/develop/site/templates/footer/nav.jet) haha.

Comment (PeculiarGoat) [reply site/templates/header-banner.jet:3]: Everything in the footer nav is indented correctly.  Your `if` block should be aligned with `showBanner`.

**Review by PeculiarGoat — APPROVED** [review]
:+1: good job!

Sorry I know the indenting stuff is annoying but we've gotta get this repo tidied up (especially since it's public).

**Pull request merged**